test/core/tsi/ssl_transport_security_utils_test.cc

```diff
@@ -429,6 +447,216 @@ INSTANTIATE_TEST_SUITE_P(FrameProtectorUtil, FlowTest,
+  ASSERT_EQ(crl_slice.status(), absl::OkStatus()) << crl_slice.status();
+  X509_CRL* crl = ReadCrl(crl_slice->as_string_view());
+  X509* cert = nullptr;
+  EXPECT_NE(VerifyCrlCertIssuerNamesMatch(crl, cert), 0);
+}
+
+TEST(CrlUtils, VerifyIssuerNameNullCrlAndCert) {
+  X509_CRL* crl = nullptr;
+  X509* cert = nullptr;
+  EXPECT_NE(VerifyCrlCertIssuerNamesMatch(crl, cert), 0);
+}
+
+TEST(CrlUtils, VerifyCrlSignBitExists) {
+  absl::StatusOr<Slice> issuer_slice = LoadFile(kCrlIssuer, false);
+  ASSERT_EQ(issuer_slice.status(), absl::OkStatus());
+  X509* issuer = ReadPemCert(issuer_slice->as_string_view());
+  EXPECT_TRUE(VerifyCrlSignBit(issuer));
+}
+
+TEST(CrlUtils, VerifyCrlSignBitMissing) {
+  absl::StatusOr<Slice> issuer_slice = LoadFile(kLeafCert, false);
+  ASSERT_EQ(issuer_slice.status(), absl::OkStatus());
+  X509* issuer = ReadPemCert(issuer_slice->as_string_view());
+  EXPECT_FALSE(VerifyCrlSignBit(issuer));
+}
+
+TEST(CrlUtils, VerifyCrlSignBitNullCert) {
+  X509* issuer = nullptr;
+  EXPECT_FALSE(VerifyCrlSignBit(issuer));
+}
+
+TEST(CrlUtils, IssuerFromIntermediateCert) {
+  absl::StatusOr<Slice> cert_slice = LoadFile(kIntermediateCrlIssuer, false);
+  ASSERT_EQ(cert_slice.status(), absl::OkStatus());
+  X509* cert = ReadPemCert(cert_slice->as_string_view());
+  auto issuer = IssuerFromCert(cert);
+
+  // Build the known name for comparison
+  unsigned char* buf = nullptr;
+  X509_NAME* expected = X509_NAME_new();
+  ASSERT_TRUE(X509_NAME_add_entry_by_txt(
+      expected, "C", MBSTRING_ASC, (const unsigned char*)"AU", -1, -1, 0));
+  ASSERT_TRUE(X509_NAME_add_entry_by_txt(expected, "ST", MBSTRING_ASC,
+                                         (const unsigned char*)"Some-State", -1,
+                                         -1, 0));
+  ASSERT_TRUE(X509_NAME_add_entry_by_txt(
+      expected, "O", MBSTRING_ASC,
+      (const unsigned char*)"Internet Widgits Pty Ltd", -1, -1, 0));
+  ASSERT_TRUE(X509_NAME_add_entry_by_txt(
+      expected, "CN", MBSTRING_ASC, (const unsigned char*)"testca", -1, -1, 0));
+  int len = i2d_X509_NAME(expected, &buf);
+  std::string expected_name(reinterpret_cast<char const*>(buf), len);
+  OPENSSL_free(buf);
+  X509_NAME_free(expected);
+
+  ASSERT_EQ(issuer.status(), absl::OkStatus());
+  std::cout << *issuer << "xxx\n";
+  EXPECT_EQ(*issuer, expected_name);
+}
+
+TEST(CrlUtils, IssuerFromLeaf) {
+  absl::StatusOr<Slice> cert_slice = LoadFile(kLeafCert, false);
+  ASSERT_EQ(cert_slice.status(), absl::OkStatus());
+  X509* cert = ReadPemCert(cert_slice->as_string_view());
+  auto issuer = IssuerFromCert(cert);
+
+  // Build the known name for comparison
+  unsigned char* buf = nullptr;
+  X509_NAME* expected = X509_NAME_new();
+  ASSERT_TRUE(X509_NAME_add_entry_by_txt(
+      expected, "CN", MBSTRING_ASC,
+      (const unsigned char*)"intermediatecert.example.com", -1, -1, 0));
+  int len = i2d_X509_NAME(expected, &buf);
+  std::string expected_name(reinterpret_cast<char const*>(buf), len);
```

Comment:
nit: Should we call this "expected_issuer_name_der"?

Reply:
See other comment

Reply:
Same here.

Reply:
Changed